### PR TITLE
Fix xwayland

### PR DESCRIPTION
**Pull request by wineee**

### CODE REVIEW

File: examples/tinywl/rootsurfacecontainer.cpp [hunk 1]
```diff
@@ -200,6 +201,7 @@ void RootSurfaceContainer::endMoveResize()
 
     ensureSurfaceNormalPositionValid(moveResizeState.surface);
 
+    moveResizeState.surface->setXwaylandPositionFromSurface(false);
```
Comment (zccrs) [inline examples/tinywl/rootsurfacecontainer.cpp:204]: Should true?

File: examples/tinywl/surfacewrapper.cpp [hunk 2]
```diff
@@ -76,6 +77,31 @@ SurfaceWrapper::SurfaceWrapper(QmlEngine *qmlEngine, WToplevelSurface *shellSurf
...
+            xwaylandSurfaceItem->setSurfaceWrapperPosition(position(), !m_xwaylandPositionFromSurface);
+        });
+
+        connect(this, &QQuickItem::yChanged, xwaylandSurface, [this, xwaylandSurfaceItem]() {
+            xwaylandSurfaceItem->setSurfaceWrapperPosition(position(), !m_xwaylandPositionFromSurface);
+        });
+
+        // xwaylandSurface->requestConfigureFlags().testAnyFlags(WXWaylandSurface::ConfigureFlag::XCB_CONFIG_WINDOW_POSITION)
+        auto requestPos = xwaylandSurface->requestConfigureGeometry().topLeft();
+        if (!requestPos.isNull()) { // howto check WINDOW_POSITION never configured?
```
Comment (zccrs) [inline examples/tinywl/surfacewrapper.cpp:100]: Use `ConfigureFlag`

File: examples/tinywl/surfacewrapper.cpp [hunk 3]
```diff
@@ -76,6 +77,31 @@ SurfaceWrapper::SurfaceWrapper(QmlEngine *qmlEngine, WToplevelSurface *shellSurf
         m_surfaceItem->setFocusPolicy(Qt::NoFocus);
 #endif
     }
+
+    if (type == Type::XWayland) {
+        auto xwaylandSurface = qobject_cast<WXWaylandSurface *>(shellSurface);
+        auto xwaylandSurfaceItem = qobject_cast<WXWaylandSurfaceItem*>(m_surfaceItem);
+
+        xwaylandSurface->safeConnect(&WXWaylandSurface::requestConfigure, this, [this, xwaylandSurfaceItem]() {
```
Comment (zccrs) [inline examples/tinywl/surfacewrapper.cpp:85]: Can use `implicitPositionChanged` from WXWaylandSurfaceItem?

**Review by zccrs — CHANGES_REQUESTED** [review]

File: examples/tinywl/surfacewrapper.cpp (hunk 2 above)
Comment (wineee) [reply examples/tinywl/surfacewrapper.cpp:100]: `requestConfigureFlags` will be override if config twice

Comment (zccrs) [reply examples/tinywl/surfacewrapper.cpp:100]: OK, remvoe the comment

**Review by zccrs — CHANGES_REQUESTED** [review]

**Review by zccrs — APPROVED** [review]

**Pull request merged**